bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiConnector.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.smhi.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static org.openhab.binding.smhi.internal.SmhiBindingConstants.*;
+
+/**
+ * Class for handling http requests to Smhi's API and return values.
+ *
+ * @author Anders Alfredsson - Initial contribution
+ */
+@NonNullByDefault
+public class SmhiConnector {
+
+    private final Logger logger = LoggerFactory.getLogger(SmhiConnector.class);
+
+    private static final String ACCEPT = "application/json";
+
+    private final HttpClient httpClient;
+
+    public SmhiConnector(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Get the reference time (the time when the forecast starts) of the latest published forecast
+     * @return A {@link ZonedDateTime} with the time of the latest forecast.
+     */
+    public ZonedDateTime getReferenceTime() {
+        logger.debug("Fetching reference time");
+        Request req = httpClient.newRequest(APPROVED_TIME_URL);
+        req.accept(ACCEPT);
+        ContentResponse resp;
+        try {
+            resp = req.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new SmhiException();
```

Comment:
Didn't know about checked vs unchecked exceptions, thanks for enlightening me :). Would altering `SmhiException` to extend `Exception` instead of `RuntimeException` be better?

Reply:
That would be a checked Exception, then. So, yes, this would be better!